src/enigma2/TimeshiftBuffer.cpp

```diff
@@ -39,6 +39,16 @@ TimeshiftBuffer::~TimeshiftBuffer(void)
   }
   if (m_filebufferReadHandle)
     XBMC->CloseFile(m_filebufferReadHandle);
+  
+  try
+  {
+    XBMC->DeleteFile(m_bufferPath.c_str());
+  }
+  catch (...)
```

Comment:
Are you sure that `DeleteFile` actually throws an exception if something goes wrong? 

https://github.com/xbmc/xbmc/blob/master/xbmc/addons/kodi-addon-dev-kit/include/kodi/libXBMC_addon.h#L499

Method signature does no contain a throw clause. Error seems to be propagated using the `bool` return value only.